File: net/Azure.Test.PerfStress/SleepTest.cs
using System;
using System.Numerics;
using System.Threading;
using System.Threading.Tasks;

namespace Azure.Test.PerfStress
{
    // Used for verifying the perf framework correctly computes average throughput across parallel tests of different speed
    public class SleepTest : PerfStressTest<PerfStressOptions>
    {
        private static int _instanceCount = 0;
        private readonly int _secondsPerOperation;

        public SleepTest(PerfStressOptions options) : base(options) {
            // Each instance of this test completes operations at a different rate, to allow for testing scenarios where
            // some instances are still waiting when time expires.  The first instance completes in 2 seconds per operation,
            // the second instance in 4 seconds, the third instance in 8 seconds, and so on. 

            var instanceCount = Interlocked.Increment(ref _instanceCount);
            _secondsPerOperation = Pow(2, instanceCount);
        }

        private static int Pow(int value, int exponent)
        {
            return (int)BigInteger.Pow(new BigInteger(value), exponent);
        }

        public override void Run(CancellationToken cancellationToken)
        {
            Thread.Sleep(TimeSpan.FromSeconds(_secondsPerOperation));
        }

        public override Task RunAsync(CancellationToken cancellationToken)
        {
            return Task.Delay(TimeSpan.FromSeconds(_secondsPerOperation), cancellationToken);
        }
    }
}

File: sdk/core/azure-core-test-perf/src/main/java/com/azure/core/test/perf/SleepTest.java
package com.azure.perfstress;

import reactor.core.publisher.Mono;

import java.time.Duration;
import java.util.concurrent.atomic.AtomicInteger;

public class SleepTest extends PerfStressTest<PerfStressOptions> {
    private static final AtomicInteger _instanceCount = new AtomicInteger();
    private final int _secondsPerOperation;

    public SleepTest(PerfStressOptions options) {
        super(options);

        int instanceCount = _instanceCount.incrementAndGet();
        _secondsPerOperation = Pow(2, instanceCount);
    }

    private static int Pow(int value, int exponent) {
        int power = 1;
        for (int i=0; i < exponent; i++) {
            power *= value;
        }
        return power;
    }

    @Override
    public void Run() {
        try {
            Thread.sleep(_secondsPerOperation * 1000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public Mono<Void> RunAsync() {
        return Mono.delay(Duration.ofSeconds(_secondsPerOperation)).then();
    }
}